### DISCUSSION TITLE
dump-disqus-for-giscus/

### DISCUSSION THREAD

**Discussion by giscus** [ironicbadger/blog.ktz.me#213] [Announcements]
# dump-disqus-for-giscus/

This is an easy project. Sure it will probably take an hour or two to get your head around but don't put it off. The end result is great and means no ads or tracking are insidiously injected into your content anymore.

https://blog.ktz.me/dump-disqus-for-giscus/

**Comment by sebasptsch** [comment]
Wow, this looks awesome. Is it worth self-hosting?

**Reply by ironicbadger** [reply]
What would you self host though? GitHub? I guess the next step would be something Gitlab or Gitea integration but what makes Giscus so powerful in my opinion, is that you're leveraging the power of community. 

Most people reading a technical blog like mine likely already have a GitHub account. It is more unlikely that a spammer would though isn't it?

It, like most things in life, is a balancing act of pragmatism vs idealism. So far so good on the pragmatism-o-tron from me!

**Reply by sebasptsch** [reply]
I just saw a subheading on their docs that said self-hosting. After reading into it a bit more it's not self-hosting self-hosting but rather get vercel to host it for you.

**Reply by curtisspendlove** [reply]
So, self-hosting would basically just be hosting the `giscuss` app rather than GitHub. It would still use GitHub behind the scenes, but you’d basically be hosting the `giscuss.app` app / script on your own infrastructure. And then you’d authenticate your own version of that app instead of the officially-hosted one. You could also then refer to your hosted app `giscuss.ktz.me` instead of `giscuss.app`. 

Hopefully that makes sense.

**Comment by d4n3sh** [comment]
Just made the same change to my hugo blog :D

**Comment by StarrWulfe** [comment]
This is pretty nice— and maybe there’s a method that can be added for [webmentions](https://www.w3.org/TR/webmention/) and some light ActivityPub too.

**Comment by curtisspendlove** [comment]
This is pretty amazing. Glad you did this and highlighted it. I’ve been wanting to revamp my defunct site and been playing with my ultimate stack. :)
